src/lib/tableland-calls.ts

```diff
@@ -141,4 +144,35 @@ async function receipt(
   return undefined;
 }
 
-export { hash, list, receipt, read, validateWriteQuery, write };
+async function setController(
+  this: Connection,
+  controller: string,
+  name: string,
+  caller?: string
+): Promise<WriteQueryResult> {
+  const parts = name.split("_");
+  if (parts.length !== 3) {
```

Comment:
I don't see the arg `name` being used here, and If the prefix has an `_` the method call would fail.  
Am I missing something?

Reply:
Name is just the table name, and we want to extract the prefix from it. I guess, since we really _only_ need prefix (chainId can be derived) we could just use a prefix arg?

Reply:
```suggestion
  if (parts.length < 3) {
```

Reply:
changing to `parts.length < 3` still won't end up making parts[1] be the chainId.  I was working on a fix for #76 that checks the network. Once #78 is done, maybe we could use that?

Reply:
yes duh, of course, good call.